app/services/Metrics.scala

```diff
@@ -1,26 +1,32 @@
 package services
 
 import metrics.{CloudWatch, CloudWatchMetrics}
+import org.apache.pekko.actor.ActorSystem
 import play.api.inject.ApplicationLifecycle
 import play.api.{Environment, Mode}
-import rx.lang.scala.Observable
+import play.api.libs.concurrent.Pekko
 
-import scala.concurrent.Future
-import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.duration.*
 
 class Metrics(
     cloudWatch: CloudWatch,
     shouldCreateMetrics: Boolean,
     namespace: String,
     securityHqNamespace: String,
     agents: Agents,
-    lifecycle: ApplicationLifecycle
+    lifecycle: ApplicationLifecycle,
+    system: ActorSystem
 ) {
+  implicit val ec: ExecutionContext = system.dispatcher
+
   if (shouldCreateMetrics) {
 
-    val subscription = Observable
-      .interval(initialDelay = 10.seconds, period = 6.hours)
-      .subscribe { _ =>
+    val subscription = system.scheduler.scheduleAtFixedRate(
```

Comment:
One gotcha here is to check what happens if you shut the laptop for a few hours (e.g. an evening) and then open it again. Does it try and catch up all the missed invocations, causing a massive pile of API requests all at once? Or does it pick up from where it is when it wakes? The old Observable approach was safe on this point, I don't know off the top of my head how a Pekko actor schedule will work!

I don't think this is a massive deal, since no one really works on this app anymore* so this is more a point of interest than a request for us to dive into it now 😄

Reply:
It should only start one instance!

The way `scheduleAtFixedRate` works under the hood is that after the Runnable completes it will schedule the next run.

I would have kept Observabale, but the library its from (rxscala) appears to be unmaintained now and doesn't have a Scala 3 version.